client/data/emails/use-remove-titan-mailbox-mutation.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+import { useMutation, useQueryClient } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import { queryKeyPrefix } from './use-emails-query';
+import wp from 'calypso/lib/wp';
+
+const noop = () => {};
+
+/**
+ * Deletes a mailbox from a Professional Email (Titan) account
+ *
+ * @param {string} domainName The domain name of the mailbox
+ * @param {string} mailboxName The mailbox name
+ * @param {object} mutationOptions Mutation options passed on to `useMutation`
+ * @returns {{ data, error, isLoading: boolean, removeTitanMailbox: Function, ...}} Returns various parameters piped from `useMutation`
+ */
+export function useRemoveTitanMailboxMutation( domainName, mailboxName, mutationOptions = null ) {
+	mutationOptions = mutationOptions ?? {};
+
+	const suppliedOnSettled = mutationOptions.onSettled ?? noop;
+
+	const queryClient = useQueryClient();
+
+	mutationOptions.onSettled = ( data, error, variables, context ) => {
```

Comment:
Why do we need to define these mutations options? I'm actually not sure what `onSettled` means. It doesn't seem necessary looking at the other places in Calypso that define a mutation for React Query ([an example](https://github.com/Automattic/wp-calypso/blob/d64f25956d54275c213f948b3e06180267d73482/client/data/users/use-update-user-mutation.js#L13)), and which actually look simpler.

Reply:
Please take a look at the definition of the `onSettled` feature. This utilisation is not so different from other examples in Calypso, even though the solutions are not necessarily similar.